### PR TITLE
Add schedule flow toasts and closed posting state

### LINKED ISSUE #524: Polish Admin Homepage View Schedule Flow
Make existing behaviour of admin homepage schedule flow compliant with https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1421%3A39619

Missing toasts:

Extending closing dates case (Edit: Note doing since cut from scope):

### PR DESCRIPTION
**Pull request by MatoPlus**
## Ticket link

Closes #524

NOTE: This ticket does not include the "reopen posting" modal logic as it is no longer valid. 

## Implementation description
* Upgraded chakra/react to 1.7.3 to allow styling of toasts for the error toast that shows up on postings that has not closed yet
  * Updated usage of removed props such as "align" on Box/Container components in newer version of chakra
* Added missing success toast for publishing post, drafting post
* Added state for postings with auto closing date not passed yet (admin should not be able to schedule volunteer before the posting signup date has passed)

## Steps to test
1. When viewing a posting as an admin with it's auto closing date not passed yet, a persistent error toast should pop up at the bottom to inform user they cannot schedule volunteers until the date has passed
2. When creating/editing postings as an admin, upon success, an success toast should show up, informing user that they have made the change to the posting

## What should reviewers focus on?
* Code correctness for new business logic 
* Code cleanliness

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/schedule/AdminScheduleVolunteerTable.tsx [hunk 1]
```diff
@@ -81,9 +84,9 @@ const AdminScheduleVolunteerTable = ({
             <Text
               textStyle="button-semibold"
               fontSize="12px"
-              color="violet"
+              color={isEditing ? "violet" : "purple.600"}
```
Comment (neelismail01) [inline frontend/src/components/admin/schedule/AdminScheduleVolunteerTable.tsx:87]: Can we add this colour definition into our theme?

Comment (MatoPlus) [reply frontend/src/components/admin/schedule/AdminScheduleVolunteerTable.tsx:87]: I don't really see value in that to be honest. Are you talking about "purple.600" that is also the name we refer to in the figma design so I think it's fine.

**Comment by MatoPlus** [comment]
Looks like I got some lint errors to take care of. I'll try to get them in by EOD

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx [hunk 2]
```diff
@@ -131,6 +136,8 @@ const ShiftScheduleCalendar = ({
   );
 
 const SchedulePostingPage = (): React.ReactElement => {
+  const toast = useToast();
```
Comment (MatoPlus) [inline frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:139]:
```suggestion
```

File: frontend/src/components/pages/admin/posting/PostingFormReviewPage.tsx [hunk 3]
```diff
@@ -55,6 +55,7 @@ const PostingFormReviewPage = ({
   title,
 }: PostingFormPageProps): React.ReactElement => {
   const { branch, skills, employees, ...rest } = useContext(PostingContext);
+  const toast = useToast();
```
Comment (MatoPlus) [inline frontend/src/components/pages/admin/posting/PostingFormReviewPage.tsx:58]:
```suggestion
```

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx [hunk 4]
```diff
@@ -377,13 +393,39 @@ const SchedulePostingPage = (): React.ReactElement => {
       ),
     ) === PostingFilterStatus.PAST;
 
+  // TODO: If posting is closed, we want to disable action to review registrants
```
Comment (MatoPlus) [inline frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:396]:
```suggestion
```

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx [hunk 5]
```diff
@@ -377,13 +393,38 @@ const SchedulePostingPage = (): React.ReactElement => {
...
   useEffect(() => {
     return () => {
       setVolunteerId("");
       setDisplayVolunteerSidePanel(false);
     };
   }, []);
 
+  useEffect(() => {
+    if (!isPostingClosed && postingDetails !== undefined) {
+      toast({
```
Comment (sherryhli) [inline frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:409]: Might be nice to add some logic for avoiding duplicate toasts, which can occur if returning to the schedule page after navigating away (e.g. clicking "View Posting" and then "Back")

https://chakra-ui.com/docs/components/toast#preventing-duplicate-toast

image

Comment (sherryhli) [inline frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:409]: Additionally, I'm seeing something strange with 2 toasts appearing on subsequent visits to the same schedule page 🤔

schedule-toast

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx [hunk 6]
```diff
@@ -377,13 +393,38 @@ const SchedulePostingPage = (): React.ReactElement => {
...
+        title:
+          "Changes to this page cannot be made until posting closing date has passed",
+        status: "error",
+        position: "bottom-left",
+        duration: null,
+        isClosable: true,
+        variant: "subtle",
+        containerStyle: {
+          minWidth: "75vw",
```
Comment (sherryhli) [inline frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:418]: Should the toast width match the calendar width?

**Review by sherryhli — CHANGES_REQUESTED** [review]
Noticed a couple of possible bugs on the "closing date not passed" toast

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx (hunk 5 above)
Comment (MatoPlus) [reply frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:409]: Good catch. I'll investigate

File: frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx (hunk 6 above)
Comment (MatoPlus) [reply frontend/src/components/pages/admin/schedule/SchedulePostingPage.tsx:418]: I think it should yea. Let me see if I can make that change.

**Review by MatoPlus — COMMENTED** [review]
resolved @sherryhli suggestions, still kinda bothered by how the toast in question persist even after we leave the page but I think that's just how toasts are programmed. I'll just put a 10s timer on it.

**Review by sherryhli — APPROVED** [review]
LGTM code-wise

> resolved @sherryhli suggestions, still kinda bothered by how the toast in question persist even after we leave the page but I think that's just how toasts are programmed. I'll just put a 10s timer on it.

Yeah, not sure what the design intent is here, but perhaps a custom-built persistent banner on the schedule page might be more appropriate than a toast. Understandable if there's no capacity to address this though.

**Comment by MatoPlus** [comment]
> LGTM code-wise
> 
> > resolved @sherryhli suggestions, still kinda bothered by how the toast in question persist even after we leave the page but I think that's just how toasts are programmed. I'll just put a 10s timer on it.
> 
> Yeah, not sure what the design intent is here, but perhaps a custom-built persistent banner on the schedule page might be more appropriate than a toast. Understandable if there's no capacity to address this though.

Yea I think you're right. I'll look other options for components

**Pull request merged**